src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -256,10 +254,10 @@ IntPtr LocalObjectToHandle(object input)
             {
                 fixed(byte* pInstrumentationData = &instrumentationData[0])
                 {
-                    LikelyClassRecord* likelyClasses = stackalloc LikelyClassRecord[MAX_LIKELY_CLASSES];
-                    uint numberOfClasses = getLikelyClasses(likelyClasses, MAX_LIKELY_CLASSES, pSchema, 2, pInstrumentationData, nativeSchema[index].ILOffset);
-
                     // We're going to store only the most popular type to reduce size of the profile
+                    LikelyClassRecord* likelyClasses = stackalloc LikelyClassRecord[1];
+                    uint numberOfClasses = getLikelyClasses(likelyClasses, 1, pSchema, 2, pInstrumentationData, nativeSchema[index].ILOffset);
+
```

Comment:
Since this will never return null handles, we should no longer need to check the result against `IntPtr.Zero` right?